SCCS analysis

svetlanabelitser opened the pull request:
Here is a list of all [new] files until now:

Parameters file in dir:' ../p_parameters/':
* '07_scri_inputs.R' 

Functions in directory '../p_steps/':
* 'standardsccs2' in file 'standardsccs2.R'
* 'formatdata2'  in file 'formatdata2.R'
* 'table1'   in file 'table1.R'
* mb function to create plots. At this moment just R code.

R programs in directory '../p_steps/':
* step_12_1_create_scri_dataset.R
* step_12_2_run_scri.R
* plot_sccs.R

Markdown file to create report in directory '../p_steps/':
* step_12_report_sccs.Rmd  (just started)
* Markdown_template.docx template-file

I am still working on function 'standardsccs2'  and R code 'step_12_2_run_scri.R';
also on plots and Markdown report.

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Merge pull request #1 from svetlanabelitser/svetlanabelitser-patch-1-1

Add files via upload

svetlanabelitser committed:
Merge pull request #2 from svetlanabelitser/svetlanabelitser-patch-1

Svetlanabelitser patch 1

svetlanabelitser committed:
Update step_12_1_create_scri_dataset.R

annaschultze commented on p_parameters/07_scri_inputs.R:
```diff
@@ -0,0 +1,30 @@
+
+# package list 
+packages<-  c( "survival" )   
+#Sv packages<- c("tidyverse", "data.table", "here", "SCCS", "gtsummary", "janitor", "ggplot2")
```
if these are not used suggest remove

svetlanabelitser replied:
agree. Then I would add "SCCS". 
packages<-  c( "survival", "SCCS" )

annaschultze commented on p_steps/standardsccs2.R:
```diff
@@ -0,0 +1,231 @@
+  mod <- try( clogit(formula = fmla1, data = chopdat) )
+  
+  if(class(mod)[[1]]== "try-error"){ 
+    warning(paste("The SCCS model could not be run. Try to repeate for groups with enough cases.\n"))
```
I wasn't sure about this error message, specifically, if it's sure that the model could not be run or just the RR for certain covariate levels could not be estimated. Could you confirm?

annaschultze replied:
I think we want to give output for this either way, it's quite common to have levels of the exposure that cannot be estimated (ie a day 0 effect, as we do here)

annaschultze replied:
suggest changing to "There are small numbers in at least one strata, check the SCCS model specification"

annaschultze commented on p_steps/standardsccs2.R:
```diff
@@ -0,0 +1,231 @@
+  
+  events_sum <- cbind.data.frame(events_sum, atrisk_days[,ncol(atrisk_days)], atrisk_persons[,ncol(atrisk_persons)])
+  names(events_sum)[ncol(events_sum)+(-2:0)] <- c("n_events","atrisk_days","atrisk_persons")
+  events_sum$event_rate    <- events_sum$n_events / events_sum$atrisk_days    * 100
```
As this is a case only design we cannot calculate event rates - the RRs are estimated indirectly through the likelihood instead and this estimate won't be meaningful. Suggest removed to not cause confusion.

annaschultze commented on p_steps/step_12_1_create_scri_dataset.R:
```diff
@@ -0,0 +1,111 @@
+scri_data_extract$dose_diff = as.numeric(difftime(scri_data_extract$date_vax2, scri_data_extract$date_vax1,units="days"))
+summary(scri_data_extract$dose_diff)
+if(any(min(scri_data_extract$dose_diff[!is.na(scri_data_extract$dose_dif)])<14)) 
+  warning(paste("There are ", sum(scri_data_extract$dose_diff<14, na.rm=T), " persons with (date_vax2 - date_vax1) < 14 days."))
```
This is a nice sense check

annaschultze commented on p_steps/step_12_1_create_scri_dataset.R:
```diff
@@ -0,0 +1,111 @@
+
+
+
+# check: vaccine dates. If (date_vax2 - date_vax1) <= 5 ==> look for next vaccine and replace. 
```
so I think this  makes sense but I thought this kind of cleaning would be done by ARS already at an earlier stage, so that things are consistent between the cohort and the SCRI. To confirm?

annaschultze replied:
@DavideMessinaARS can you confirm that cleaning of vaccine doses is done at an earlier stage?

DavideMessinaARS commented:
When you finish the code for running the basic part of the script (without plots and markdown) give here a heads-up and I will merge it to main.

Then I will run the script on ARS data.

This afternoon I will try to run until the step 12_1.

annaschultze commented on p_steps/step_12_1_create_scri_dataset.R:
```diff
@@ -0,0 +1,111 @@
+    pats_dose_diff_le5 <- scri_data_extract$person_id[ cond ]
+    scri_data_extract$dose_diff[scri_data_extract$person_id %in% pats_dose_diff_le5]
+    
+    # replace vax2 by vax3; vax3 by vaX4:
```
I'm not sure if vax3/4 will exist

annaschultze commented on p_steps/step_12_1_create_scri_dataset.R:
```diff
@@ -0,0 +1,111 @@
+# Program Information  ----------------------------------------------------
+
+# Program:      step_12_1_create_scri_dataset.R 
```
I think this should be renamed, as I think it does more data cleaning than creates a data set? I.e. inclusion/exclusion criteria are not applied here.

annaschultze commented on p_steps/step_12_1_create_scri_dataset_with_simulation.R:
```diff
@@ -0,0 +1,152 @@
+summary(scri_data_extract)
+
+
+# simulation for CPRD: 
```
I think this needs to be removed

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+summary(scri_input)
+
+#####
+# change time scale to days since 01-01-19                #??????????
```
not sure why this is required?

annaschultze replied:
Agreed to change to Sep 1 as per SAP

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+#####  define lengths of the risk windows  #####
+#
+# define length of pre-vaccin. period:
+prevax_per_start <- -92   # let op negative sign!!!
```
90? why 92?

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+######################################################################
+# define start and end of observation
+#   start0,end0 - all obs period
+#   start,end   - 1y before vax 1 and 1y after vax 2
```
not sure why these are set as the year before/after?

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+#   start0,end0 - all obs period
+#   start,end   - 1y before vax 1 and 1y after vax 2
+scri_input$start   <- pmax( scri_input$study_entry_days, scri_input$days_vax1 + prevax_per_start, na.rm=T)   #?
+scri_input$end     <- pmin( scri_input$study_exit_days,  scri_input$days_vax2 + vax2_end,         na.rm=T)   #? 
```
what is end for those who don't have vax2?

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+SCCS_models_A <- list()
+
+scri_input$vd <- scri_input$vd0
+res<-standardsccs2(  event ~ vd, 
```
I think we only want analyses per brand, so this can probably be removed

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+# ensure required folders are created  
+dir.create(file.path("./g_intermediate/scri"), showWarnings = FALSE, recursive = TRUE)
+dir.create(file.path("./g_output/scri"), showWarnings = FALSE, recursive = TRUE)
+dir.create(file.path("./log_files/scri"), showWarnings = FALSE, recursive = TRUE)
```
I didn't think we were allowed to output/release log files? If we are agree we should

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+dir.create(file.path("./log_files/scri"), showWarnings = FALSE, recursive = TRUE)
+
+# SCCS output_directory  
+sdr <- paste0(getwd(),"/g_output/scri/SCCS/")
```
unsure why this separate directory is required, think we should just use the scri one (particularly as it isn't an SCCS to not confuse things?)

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+scri_input$age_start_study_cat <- cut(scri_input$age_start_study,age_cat)
+table1(scri_input$age_start_study_cat,title="agecat")
+
+
```
placeholder to define calendar time cut-offs for adjusted analyses?

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+
+scri_input$study_entry_date0 <- scri_input$study_entry_date
+scri_input$study_entry_date  <- pmin( scri_input$study_entry_date, scri_input$start_study_date )
+
```
need to ensure we account for CPRD end dates

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+# define start and end of observation
+#   start0,end0 - all obs period
+#   start,end   - 1y before vax 1 and 1y after vax 2
+scri_input$start   <- pmax( scri_input$study_entry_days, scri_input$days_vax1 + prevax_per_start, na.rm=T)   #?
```
I wasn't sure in what situation someone's study entry would be > their prevax start, as I think we set the entry as a calendar date at least 90 days prior to the first vaccine dose?

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+# prevax risk period: 
+scri_input$vd0     <- pmax( scri_input$study_entry_days, scri_input$days_vax1 + prevax_per_start, na.rm=T) 
+scri_input$evd0    <- pmin( scri_input$study_exit_days,  scri_input$days_vax1 + prevax_per_end,   na.rm=T) 
+scri_input$vd      <- scri_input$vd0
```
unclear on difference between vd and vd0

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+prevax_per_end   <- -30   # let op negative sign!!!
+#
+# define risk period of vax 1:
+vax1_start <- 0  
```
I think we always want to treat day 0 separately, so this should be 1?

annaschultze commented on p_steps/step_12_2_run_scri.R:
```diff
@@ -0,0 +1,250 @@
+# vax 2 risk period: 
+scri_input$vd2  <- scri_input$days_vax2 + vax2_start 
+scri_input$evd2 <- pmin( scri_input$days_vax2 + vax2_end, scri_input$study_exit_days) 
+
```
I'm not clear how time between dose 1 and dose 2 risk windows is currently handled, it's important that this is not put towards control time but handled as a separate level of the exposure variable

annaschultze commented:
@DavideMessinaARS we're meeting at 15.30 EU time to resolve the review questions, I would hold off running until after that point as I'm not sure which step_12_1 should be considered/if it's finalised.

svetlanabelitser committed:
Update step_12_1_create_scri_dataset.R

svetlanabelitser committed:
Update 07_scri_inputs.R

svetlanabelitser committed:
Update table1.R

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Delete formatdata2.R

svetlanabelitser committed:
Delete standardsccs2.R

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Update step_12_1_create_scri_dataset.R

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Add files via upload

svetlanabelitser committed:
Update step_12_2_run_scri.R

svetlanabelitser committed:
Update step_12_1_create_scri_dataset.R

svetlanabelitser committed:
Update scri.R

svetlanabelitser committed:
Update step_12_2_run_scri.R

svetlanabelitser committed:
Update step_12_2_run_scri.R

svetlanabelitser committed:
Delete formatdata2.R

svetlanabelitser committed:
Delete table1.R

svetlanabelitser committed:
Delete table1.R

svetlanabelitser committed:
Delete step_12_report_sccs.Rmd

svetlanabelitser committed:
Delete step_12_1_create_scri_dataset_with_simulation.R

svetlanabelitser commented:
The scripts are ready. I changed three files:

    - p_steps/step_12_1_create_scri_dataset.R
    - p_steps/step_12_2_run_scri.R.
    - p_macro/scri.R

svetlanabelitser committed:
Update scri.R

svetlanabelitser commented:
Update scri.R :
the continuity correction is added.

annaschultze commented:
> 

Sounds good on the programs being done! For my own understanding, why did you add this correction?

svetlanabelitser committed:
Update step_12_2_run_scri.R

svetlanabelitser committed:
Add files via upload

svetlanabelitser commented:
> Sounds good on the programs being done! For my own understanding, why did you add this correction?

 To get a stratified SCCS analysis, we can run the function for each subgroup separately, or simply add the corresponding covariate into the SCCS function and get all the results in one time. For the second option, if there are no observations in one of the categories/subpopulations/strata, we get an error and no results, also not for other categories. To fix this porblem, I created observations for such categories with zero events and 0.00000001 at risk. This is similar to "continuity correction", however, not exactly the same because I only add it to empty categories, and not to all categories. After adding these nep-data, the function works without the error, and the coefficients for categories without nep-data are estimated.

svetlanabelitser committed:
Add files via upload

completely new functions for SCRI analysis

svetlanabelitser committed:
Add files via upload

step_12_1_create_scri_dataset.R is changed a little.

step_12_2_run_scri.R is almost completely changed. From now on, only functions from 'scri_tool.R' are needed! Another approach than before.

DavideMessinaARS committed:
Merge branch 'SCCS' into main
